lib/property_constraints.js

```diff
@@ -1,95 +1,116 @@
 var Immutable = require('immutable');
 
-var cardinality = {
-  '=': function(val, count) {
-    return val.size === parseInt(count);
-  },
-  '>': function(val, count) {
-    return val.size > parseInt(count);
-  },
-  '<': function(val, count) {
-    return val.size < parseInt(count);
-  }
+function not(fn) {
+  return function(state) {
+    return !fn(state);
+  };
+}
+
+function and() {
+  var fns = Array.prototype.slice.call(arguments);
+  return function(state) {
+    return fns.reduce(function(memo, fn) {
+      return memo && fn(state);
+    }, true);
+  };
+}
+
+
+var constraint = function() {
```

Comment:
I think pipeline makes sense in conceptually what it is. But in a DSL I prefer using a context specific term.